src/Main.elm

```diff
@@ -1,16 +1,411 @@
 module Main exposing (main)
 
 import Html exposing (Html)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import FontAwesome.Web as Icon
 
 
 -- Import Modules
 
-import Data.Audience
-import Data.AudienceFolder
+import Data.Audience exposing (Audience, AudienceType(..))
+import Data.AudienceFolder exposing (AudienceFolder, audienceFoldersJSON)
+import Json.Decode as Decode exposing (Decoder, field)
+import Json.Decode.Extra as Decode exposing ((|:))
 
 
 {-| Main file of application
 -}
-main : Html msg
+main : Program Never Model Msg
 main =
-    Html.text "There will be app soon!"
+    -- Html.text "There will be app soon!"
+    Html.program
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { audienceFolders : List AudienceFolder
+    , audiences : List Data.Audience.Audience
+    , currentPath : List AudienceFolder
+    , selectedFolder : Maybe AudienceFolder
+    }
+
+
+init : ( Model, Cmd Msg )
+init =
+    let
+        audienceFolder =
+            case Debug.log "audiencefolder" (audienceFoldersDecoder audienceFoldersJSON) of
+                Ok data ->
+                    data
+
+                -- @TODO: error reporting
+                Err err ->
+                    []
+
+        audiences =
+            case Debug.log "audiences" (audiencesDecoder Data.Audience.audiencesJSON) of
+                Ok data ->
+                    data
+
+                -- @TODO: error reporting
+                Err err ->
+                    []
+    in
+        ( { audienceFolders = audienceFolder
+          , audiences = audiences
+          , currentPath = []
+          , selectedFolder = Nothing
+          }
+        , Cmd.none
+        )
+
+
+
+-- DECODERS
+
+
+audienceFoldersDecoder : String -> Result String (List AudienceFolder)
+audienceFoldersDecoder json =
+    json
+        |> Decode.decodeString (Decode.at [ "data" ] <| Decode.list audienceFolderDecodeItem)
+
+
+audienceFolderDecodeItem : Decoder AudienceFolder
+audienceFolderDecodeItem =
+    Decode.succeed AudienceFolder
+        |: (field "id" Decode.int)
+        |: (field "name" Decode.string)
+        |: (field "parent" <| Decode.maybe Decode.int)
+
+
+audiencesDecoder : String -> Result String (List Audience)
+audiencesDecoder json =
+    json
+        |> Decode.decodeString (Decode.at [ "data" ] <| Decode.list audiencesDecoderItem)
```

Comment:
`Decode.decodeString (Decode.at [ "data" ] <| Decode.list` can be abstracted to own function `decodeCollection`